tests/unit/serve/runtimes/worker/test_worker_runtime.py

```diff
@@ -275,5 +274,69 @@ async def task_wrapper(adress, messages_received):
     assert not WorkerRuntime.is_ready(f'{args.host}:{args.port}')
 
 
+class SlowInitExecutor(Executor):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        time.sleep(5.0)
+
+    @requests
+    def foo(self, docs, **kwargs):
+        return docs
+
+
+@pytest.mark.timeout(10)
+@pytest.mark.asyncio
+async def test_worker_runtime_slow_init_exec():
+    args = set_pod_parser().parse_args(['--uses', 'SlowInitExecutor'])
+
+    cancel_event = multiprocessing.Event()
+
+    def start_runtime(args, cancel_event):
+        with WorkerRuntime(args, cancel_event) as runtime:
+            runtime.run_forever()
+
+    runtime_thread = Process(
+        target=start_runtime,
+        args=(args, cancel_event),
+        daemon=True,
+    )
+    runtime_started = time.time()
+    runtime_thread.start()
+
+    # wait a bit to the worker runtime has a chance to finish some things, but not the Executor init (5 secs)
+    time.sleep(1.0)
+
+    # try to connect a TCP socket to the gRPC server
+    # this should only succeed after the Executor is ready, which should be after 5 seconds
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    connected = False
+    while not connected:
+        try:
+            s.connect((args.host, args.port))
+            connected = True
+        except ConnectionRefusedError:
+            time.sleep(0.2)
```

Comment:
I had some issue with pytest when I don't close the socket. Here it might not change anything but it might prevent future but in the test maybe
```suggestion
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        connected = False
        while not connected:
            try:
                s.connect((args.host, args.port))
                connected = True
            except ConnectionRefusedError:
                time.sleep(0.2)
```

Reply:
good point, added it!